hw_6/main.py

```diff
@@ -0,0 +1,196 @@
+import databases
+import sqlalchemy
+from fastapi import FastAPI
+from pydantic import BaseModel
+from typing import List
+import datetime
+
+DATABASE_URL = "sqlite:///./test.db"
+database = databases.Database(DATABASE_URL)
+metadata = sqlalchemy.MetaData()
+
+users = sqlalchemy.Table(
+    "users",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String),
+    sqlalchemy.Column("surname", sqlalchemy.String),
+    sqlalchemy.Column("email", sqlalchemy.String),
+    sqlalchemy.Column("password", sqlalchemy.String),
+)
+
+products = sqlalchemy.Table(
+    "products",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String),
+    sqlalchemy.Column("description", sqlalchemy.String),
+    sqlalchemy.Column("price", sqlalchemy.Integer),
+)
+
+orders = sqlalchemy.Table(
+    "orders",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("user_id", sqlalchemy.Integer, sqlalchemy.ForeignKey("users.id")),
+    sqlalchemy.Column("product_id", sqlalchemy.Integer, sqlalchemy.ForeignKey("products.id")),
+    sqlalchemy.Column("order_date", sqlalchemy.DateTime),
+    sqlalchemy.Column("status", sqlalchemy.String),
+)
+
+engine = sqlalchemy.create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
+metadata.create_all(engine)
+
+app = FastAPI()
+
+
+@app.on_event("startup")
+async def startup():
+    await database.connect()
+
+
+@app.on_event("shutdown")
+async def shutdown():
+    await database.disconnect()
+
+
+class UserIn(BaseModel):
+    name: str
+    surname: str
+    email: str
+    password: str
+
+
+class UserOut(BaseModel):
+    id: int
+    name: str
+    surname: str
+    email: str
+
+
+class ProductIn(BaseModel):
+    name: str
+    description: str
+    price: int
+
+
+class ProductOut(BaseModel):
+    id: int
+    name: str
+    description: str
+    price: int
+
+
+class OrderIn(BaseModel):
+    user_id: int
+    product_id: int
+    order_date: datetime.datetime
+    status: str
+
+
+class OrderOut(BaseModel):
+    id: int
+    user_id: int
+    product_id: int
+    order_date: datetime.datetime
+    status: str
```

Comment:
Модели, кстати, мало чем отличимы, следовательно, одна может наследоваться от другой.
```suggestion
class OrderIn(BaseModel):
    user_id: int
    product_id: int
    order_date: datetime.datetime
    status: str


class OrderOut(OrderIn):
    id: int
```